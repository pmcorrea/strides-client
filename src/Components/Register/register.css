.registration-form>div>button, .registration-form>div>a {
	width: 45%;
	margin: 5px 0px;
	color: var(--custom-darkblue);
	background-color: transparent;
	font-size: var(--button-font-size);
	font-weight: 700;
	padding: 5px;
	border: solid var(--custom-darkblue) 2px;
	border-radius: 15px;
	text-align: center;
	cursor: pointer;
}

.registration-form>div>button:hover, .registration-form>div>a:hover {
	color: var(--custom-white);
	background-color: var(--custom-darkblue);
	border: solid var(--custom-darkblue) 2px;
}

.registration-form>a:visited {
	color: white;
}



.registration-form>div>button:disabled {
	border-color: lightgray;
	color: lightgray;
}

.password-reqs {
	width: 200px;
	margin: 10px 0px;
}
.password-reqs>p{
	font-weight: 300;
	font-size: .8rem;
}