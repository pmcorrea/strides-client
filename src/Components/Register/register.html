<!DOCTYPE html>
<html lang="en">

	<head>
		<meta charset="UTF-8">
		<meta name="viewport" content="width=device-width, initial-scale=1.0">
		<link rel="stylesheet" href="../Login/index.css" type="text/css">
		<link href="https://fonts.googleapis.com/css2?family=Roboto:wght@100;300;400;500;700;900&display=swap" rel="stylesheet">
		<title>Strides</title>
	</head>

	<body>
		<h1>Register</h1>

		<form action="">
			<input type="text" placeholder="username">
			<input type="password" placeholder="password">
			<input type="password" placeholder="confirm password">

			<div>
				<button>Go Back</button>
				<button>Register</button>
			</div>
		</form>

	</body>

</html>