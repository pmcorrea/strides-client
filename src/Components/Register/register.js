import React, { useState } from 'react';
import { Link } from "react-router-dom";

import { useQuery, useMutation, useLazyQuery } from '@apollo/client'
import { addUser } from '../../Queries/queries'

import "./register.css"

export default function Register(routeProps) {
	const [stateError, setError] = useState(null)
	const [stateUsername, setUsername] = useState('');
	const [statePassword, setPassword] = useState('');
	const [stateConfirmPassword, setConfirmPassword] = useState('');

	const [createUser, { loading, data, error }] = useMutation(
		addUser,
		{ 
			variables: {
				user_name: stateUsername,
				user_password: statePassword
			},
			onError: (error) => {
				console.log(error.graphQLErrors[0].message)
				setError(`${error.graphQLErrors[0].message}`)
			}
		}
	)

	if (loading || loading === undefined) {
		// console.log('loading...')
	} else if (error) {
		console.log('err:', error.graphQLErrors[0].message)	
	} else if (data) {
		console.log(data["addUser"])
		routeProps.history.push("/")		
	}


	function handleSubmit(e) {
		e.preventDefault()
		createUser()
	}

	return (
		<div>
			<h1>Register</h1>

			<form action="" onSubmit={(e) => handleSubmit(e)} className="registration-form">
				<input name="username" type="text" placeholder="username" onChange={(e) => setUsername(e.target.value)}/>
				<input name="password" type="password" placeholder="password" onChange={(e) => setPassword(e.target.value)}/>
				<input name="confirmPassword" type="password" placeholder="confirm password" onChange={(e) => setConfirmPassword(e.target.value)}/>

				<div>
					<Link to="/">
						Go Back
					</Link>
				
					<button type="submit">
							Submit	
					</button>
				</div>
		</form>

			{stateError ? <p>Something went wrong, try again...</p> : (' ')}
		</div>
	);
}