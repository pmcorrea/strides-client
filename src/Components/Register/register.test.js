import React from 'react';
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect'
import Register from './register';

afterEach(cleanup)

// Snapshot Test
it ('should take a snapshot', () => {
	const { asFragment } = render(
	<BrowserRouter>
			<Register />
	</BrowserRouter>
	)

	expect(asFragment(
		<BrowserRouter>
			<Register />
		</BrowserRouter>)
		).toMatchSnapshot()
})

// Smoke Test
it('renders without crashing', () => {
	const div = document.createElement('div')

	ReactDOM.render(
		<BrowserRouter>
			<Register />
		</BrowserRouter>,
		div
	)
	
	ReactDOM.unmountComponentAtNode(div)
})