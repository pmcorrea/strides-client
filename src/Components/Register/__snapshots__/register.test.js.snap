// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Register
    </h1>
    <form
      action=""
    >
      <input
        name="username"
        placeholder="username"
        type="text"
      />
      <input
        name="password"
        placeholder="password"
        type="password"
      />
      <input
        name="confirmPassword"
        placeholder="confirm password"
        type="password"
      />
      <div>
        <button
          type="button"
        >
          <a
            href="/"
          >
            Go Back
          </a>
        </button>
        <button
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
